wavelength-plugins/friends.js

```diff
@@ -0,0 +1,117 @@
+/*
+ * FC Friends
+ * By WGC
+ */
+'use strict';
+
+function genFC() {
+	const randomCode = () => Math.floor(Math.random() * 89999999) + 10000000;
+	let currentCodes = Db.fc.keys().map(curKey => Db.fc.get(curKey));
+	let uniqueCode = `WL${randomCode()}`;
+	while (currentCodes.some(curCode => uniqueCode === curCode)) {
+		uniqueCode = `WL${randomCode()}`;
+	}
+	return uniqueCode;
+}
+const FC = {
+	getFC: function (user) {
+		user = toId(user);
+		let fcget = Db.fc.get(user);
+		if (!fcget || fcget === null) return user.reply('You don\'t have a friendcode. Use /fc make to make one.');
```

Comment:
I'm pretty sure you meant `user.popup`.

Reply:
Also, `!fcget` returns true if it is null, so we don't need an explicit `fcget === null` check.